packages/@webex/plugin-authorization-browser-first-party/src/authorization.js

```diff
@@ -320,23 +352,29 @@ const Authorization = WebexPlugin.extend({
       return;
     }
 
-    if (this.pollingRequest) {
+    if (this.pollingTimer) {
       this.eventEmitter.emit('qRCodeLogin', {
         eventType: 'authorizationFailure',
         data: {message: 'There is already a polling request'},
       });
       return;
     }
 
-    const {device_code: deviceCode, interval = 2, expires_in: expiresIn = 300} = options;
+    const {device_code: deviceCode, expires_in: expiresIn = 300} = options;
+    let interval = options.interval ?? 2;
 
-    let attempts = 0;
-    const maxAttempts = expiresIn / interval;
+    this.pollingExpirationTimer = setTimeout(() => {
+      this.cancelQRCodePolling(false);
+      this.eventEmitter.emit('qRCodeLogin', {
+        eventType: 'authorizationFailure',
```

Comment:
the emitted event types should probably be in some enum or some other type so that client using this API can easily know what are all possible events they can get without reading the implementation.